apps/estuary/test/integration/application_start_test.exs

```diff
@@ -1,11 +1,68 @@
 defmodule Estuary.StartTest do
   use ExUnit.Case
+  use Placebo
   use Divo
+  import ExUnit.CaptureLog
+  require Logger
 
   @elsa_endpoint Application.get_env(:estuary, :elsa_endpoint)
   @event_stream_topic Application.get_env(:estuary, :event_stream_topic)
 
-  test "Topic is created when Estuary starts" do
-    assert Elsa.Topic.exists?(@elsa_endpoint, @event_stream_topic)
+  describe "Estuary.MessageHandler receives messages sent to the event stream" do
+    setup do
+      allow(Estuary.MessageHandler.handle_messages(any()),
+        return: :ack,
+        meck_options: [:passthrough]
+      )
+
+      :ok
+    end
+
+    test "Topic is created when Estuary starts" do
+      assert Elsa.Topic.exists?(@elsa_endpoint, @event_stream_topic)
+    end
+
+    test "Estuary.Message reads message from eventstream when event stream receives an event" do
+      Elsa.produce(@elsa_endpoint, @event_stream_topic, {"key", "value1"})
+
+      Process.sleep(2000)
```

Comment:
Do we for sure need this sleep?

Reply:
Yeah I caught that this morning I will make this an eventually. Thanks!